presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedQueries.java

```diff
@@ -13,14 +13,34 @@
  */
 package com.facebook.presto.tests;
 
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
 import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunner;
+import static java.lang.String.format;
 
 public class TestTpchDistributedQueries
         extends AbstractTestQueries
 {
+    private static final int maxQueryLength = 1000000;
+    public static final Map<String, String> TEST_EXTRA_PROPERTIES = ImmutableMap.of();
+    public static final Map<String, String> TEST_COORDINATOR_PROPERTIES = ImmutableMap.of("query.max-length", Integer.toString(maxQueryLength));
+
     public TestTpchDistributedQueries()
             throws Exception
     {
-        super(createQueryRunner());
+        super(createQueryRunner(TEST_EXTRA_PROPERTIES, TEST_COORDINATOR_PROPERTIES));
+    }
+
+    @Test
+    public void testTooLongQuery()
+    {
+        //  Generate a super-long query: SELECT x,x,x,x,x,... FROM (VALUES 1,2,3,4,5) t(x)
+        @Language("SQL") String longQuery = "SELECT x" + Strings.repeat(",x", 500_000) + " FROM (VALUES 1,2,3,4,5) t(x)";
+        assertQueryFails(longQuery, format("Query text length \\(%s\\) exceeds the maximum length \\(%s\\)", longQuery.length(), maxQueryLength));
```

Comment:
Instead of formatting, use the exact values in the error message. It makes it easier to verify the test is working as expected (and you don't have an identical bug in both the code and test).